risc0/circuit/rv32im/src/trace.rs

```diff
@@ -47,6 +47,12 @@ pub enum TraceEvent {
         /// Data that's been written
         region: Vec<u8>,
     },
+
+    /// A page is read for the first time in a segment
+    PageIn { cycles: u64 },
+
+    /// A page has been written to for the first time in a segment
```

Comment:
So in this case, it's the last time a page is written to that will actually be performed in the circuit.